### ISSUE TITLE
D9VK corrupts XWAYLAND server

### ISSUE THREAD

**Issue by shumvgolove** [doitsujin/dxvk#1504]
I am running [Sway](https://github.com/swaywm/sway) with `xwayland` support and have 2 monitors connected - HDMI one connected into AMD card (primary) and VGA one connected into Intel integrated card (secondary). To set my primary display I am using this little command - `xrandr --output XWAYLAND1 --primary`. For easy testing I am running `wine` and `DXVK` from Lutris. As of 1.5.0, D9VK was merged into DXVK and any DXVK version >1.5.0 with any DX9 game results in immediate crash:
```
info:  D3D9: Setting display mode: 1920x1080@0
X Error of failed request:  BadMatch (invalid parameter attributes)
  Major opcode of failed request:  140 (RANDR)
  Minor opcode of failed request:  21 (RRSetCrtcConfig)
  Serial number of failed request:  240
  Current serial number in output stream:  240
Waiting on children
All children gone
Exit with returncode 1
```
D9VK somehow corrupts XWAYLAND server (from my guessing, see above) and I am unable to run any other game (with or without DXVK). In order to fix this I need to kill the `xwayland` process and restart Lutris once again.

I am not sure if I need to provide apitrace for this specific case, but if needed I would be happy to.

### Software information
Any game that supports DirectX9

### System information
- GPU: `AMD RX590`
- Driver: `RADV`
- Wine version: doesn't matter, but I've tested with `4.21`, `5.0` and `5.2`
- DXVK version: `>1.5.0`

### Log files
- d3d9.log: [Fables_d3d9.log](https://github.com/doitsujin/dxvk/files/4314256/Fables_d3d9.log)
- full Lutris log: [test.log](https://github.com/doitsujin/dxvk/files/4314260/test.log)

**Comment by shumvgolove** [comment]
After further testing it appears that this is a wine bug. `wine 4.17` is the latest version which works perfectly fine, but all versions above throws the exact same error. However, Lutris provides wine versions with prefix `-fshack` that actually works as intended. For further information - [Lutris Wiki](https://github.com/lutris/lutris/wiki/Wine-Builds) and [Reddit thread](https://www.reddit.com/r/linux_gaming/comments/cia5ql/differences_with_lutris_wine_builds_lutris/).

Also, this is duplicate of #1267 which should be closed due to the reasons that I have already described. Sorry for the noise.

**Comment by oliverklee** [comment]
@shumvgolove Just to be sure: Do you know whether there already is a ticket for this issue https://bugs.winehq.org/ so that the Wine dev are aware of this and can fix it?

**Comment by shumvgolove** [comment]
@oliverklee Not sure there is one, but maybe [wine-staging](https://github.com/wine-staging/wine-staging) offers some `xwayland` patch? :thinking:

I think we need to fill out the issue just to be sure.
